dotnet/src/webdriver/Internal/Logging/LogContext.cs

```diff
@@ -137,6 +134,19 @@ public ILogContext SetLevel(Type issuer, LogEventLevel level)
 
         public void Dispose()
         {
+            // Dispose log handlers associated with this context
+            // if they are hot handled by parent context
+            if (Handlers != null && _parentLogContext != null && _parentLogContext.Handlers != null)
+            {
+                foreach (var logHandler in Handlers)
+                {
+                    if (!_parentLogContext.Handlers.Contains(logHandler))
+                    {
+                        (logHandler as IDisposable)?.Dispose();
+                    }
+                }
+            }
```

Comment:
I tried to refactor the highlighted code block.

```cs
foreach (var logHandler in Handlers)
{
    if (logHandler is IDisposable disposableLogHandler &&
        (_parentLogContext == null || !_parentLogContext.Handlers.Contains(logHandler)))
    {
        disposableLogHandler.Dispose();
    }
}
```

Reply:
As far as I saw `public ILogHandlerList Handlers { get; }` is read-only and cannot be null, because in constructor it is always initialized with an object: `Handlers = handlers ?? new LogHandlerList(this);`. That's why I omitted null checks.

Reply:
My code is more clear, it verifies things for `null` before acting, considering the order (to be more performant). Resolving it.